Classes/Domain/Service/NodePropertyConversionService.php

```diff
@@ -155,11 +155,15 @@ protected function convertDateTime($rawValue)
      * Convert raw value to integer
      *
      * @param mixed $rawValue
-     * @return integer
+     * @return null|integer
      */
     protected function convertInteger($rawValue)
     {
-        return (int)$rawValue;
+        if (is_numeric($rawValue)) {
+            return (int) $rawValue;
+        }
+
+        return null;
```

Comment:
Exceptions have to be caught or at least nicely presented to the editor. 

I plan it as said above for 9 with https://github.com/neos/neos-development-collection/issues/4304

> Looks good for know. In the future (Neos 9) id like to have strict validation via ajax directly in the editor. (And we could make our NodePropertyConversion thing more strict by throwing exceptions ...)

but I agree that this sudden change in behaviour might not be cool as well.

I could also think of keeping this unfixed for 7.3…8.3 and only fix it properly in 9.0 - I mean the issue was open since 2018 not super critical I suppose;)

Reply:
I think the more annoying issue (that ought to be fixed for LTS imho) is https://github.com/neos/neos-ui/issues/3157 (unsetting an optional integer value is impossible).